File: packages/fixflow_core/lib/models/requests/update_repair_request_request.dart
class UpdateRepairRequestRequest {
  final int? categoryId;
  final String? description;
  final int? preferenceType;
  final double? latitude;
  final double? longitude;
  final String? address;

  UpdateRepairRequestRequest({
    this.categoryId,
    this.description,
    this.preferenceType,
    this.latitude,
    this.longitude,
    this.address,
  });

  Map<String, dynamic> toJson() => {
        if (categoryId != null) 'categoryId': categoryId,
        if (description != null) 'description': description,
        if (preferenceType != null) 'preferenceType': preferenceType,
        if (latitude != null) 'latitude': latitude,
        if (longitude != null) 'longitude': longitude,
        if (address != null) 'address': address,
      };
}

File: FixFlow/FixFlow.Application/DTOs/Request/UpdateRepairRequestRequest.cs
using System.ComponentModel.DataAnnotations;

namespace FixFlow.Application.DTOs.Request;

public class UpdateRepairRequestRequest
{
    public int? CategoryId { get; set; }

    [StringLength(2000, MinimumLength = 10, ErrorMessage = "Opis mora imati između 10 i 2000 karaktera.")]
    public string? Description { get; set; }

    public int? PreferenceType { get; set; }

    public double? Latitude { get; set; }

    public double? Longitude { get; set; }

    [StringLength(500, ErrorMessage = "Adresa ne može biti duža od 500 karaktera.")]
    public string? Address { get; set; }
}
